### ISSUE TITLE
Create a z-index manager

### ISSUE THREAD

**Issue by tresbach** [octobercms/october#1021]
While editing something on richeditor inside a relation form popup, clicking on the "table button" renders the "insert table" popup behind the relation popup.

trying to insert a table:
![captura de tela de 2015-03-24 13 08 08](https://cloud.githubusercontent.com/assets/774244/6806434/6cda6ece-d227-11e4-8a47-79dab3a50549.png)

after closing the relation form:
![captura de tela de 2015-03-24 13 08 13](https://cloud.githubusercontent.com/assets/774244/6806435/6cddee1e-d227-11e4-9f4c-b48b3dce864e.png)

**Comment by daftspunk** [comment]
There are conflicting z-index priorities in the backend. The above is a good example of one. We need to develop a z-index manager that can assign frequencies to layered elements.

**Comment by daftspunk** [comment]
See #992 for an issue with the Color Picker

**Comment by 7talents** [comment]
For anybody who needs an immediate fix (hacky): you put write your own css file and load it in your plugins controller. The z-index of the relation modal ist 1050, so you will have to change the z-index of the <code>#redactor-modal-overlay</code>, <code>#redactor-modal-box</code> and <code>#redactor-modal</code> to something higher.

Since the redactors css gets loaded after your plugins css, you will have to tweak your own css a bit so that it overrides the original css. Since the original css uses <code>!important</code> already, you will have to use it as well, and prepend the id with body so that your css will take precedence even though it was loaded prior to the original. 

In short: this is the css that will work:
<code>body #redactor-modal-overlay, body #redactor-modal-box, body #redactor-modal{z-index:1060 !important}</code>

**Comment by tplessis** [comment]
Thx @7talents for the rapid fix. Works well.

**Comment by daftspunk** [comment]
Related issue: #1145 - Test for color picker neede

**Comment by daftspunk** [comment]
The z-indexes have been reorganised via the ["Windex" system in Storm UI](https://github.com/octobercms/october/blob/develop/modules/system/assets/ui/less/windex.variables.less
). It appears to have resolved most, if not all issues. A z-index manager based on JS may still be required in future if requirements become more complex. Downgrading this issue until then.
